jazz_codeq/test/test.js

```diff
@@ -0,0 +1,757 @@
+		});
+	});
+
+	describe('getResponseForCodeqPathString tests', () => {
+		let sandbox;
+		let event = {};
+		let config = {};
+		let data = {};
+		let query = {};
+		let getQueryStub;
+		let missingFieldsStub;
+		let fromDateStub;
+		let toDateStub;
+		let fromAfterToStub;
+		let getMetricsStub;
+
+		beforeEach(() => {
+			sandbox = sinon.createSandbox();
+			query = {
+				from: "fromdate",
+				to: "todate"
+			};
+			event = {
+				resourcePath: 'codeq',
+				method: 'GET',
+				stage: 'test',
+				query: {}
+			};
+			config = {
+				"METRICS_HELP": {
+					"security": {
+						"description": "Number of security vulnerabilities",
+						"unit": "count",
+						"minValue": "0",
+						"maxValue": "NA"
+					},
+					"code-coverage": {
+						"description": "Coverage answers the following question: 'Has each boolean expression been evaluated both to true and false?'. This is the density of possible branches in flow control structures that have been followed during unit tests execution.",
+						"unit": "percentage",
+						"minValue": "0",
+						"maxValue": "100"
+					}
+				}
+			};
+			data = { "pathString": "codeq" };
+			getQueryStub = sinon.stub(utils, "getQuery").returns(query);
+			missingFieldsStub = sinon.stub(validation, "validateMissingFields");
+			fromDateStub = sinon.stub(validation, "validateFromDate").returns('fromdate');
+			toDateStub = sinon.stub(validation, "validateToDate").returns('todate');
+			fromAfterToStub = sinon.stub(validation, "validateFromAfterTo").returns(true);
+		});
+
+		afterEach(() => {
+			sandbox.restore();
+			event = {};
+			config = {};
+			query = {};
+			getQueryStub.restore();
+			missingFieldsStub.restore();
+			fromDateStub.restore();
+			toDateStub.restore();
+			fromAfterToStub.restore();
+			getMetricsStub.restore();
+		});
+
+		it('should return error when getMetrics is called for pathString "codeq"', () => {
+			getMetricsStub = sinon.stub(utils, "getMetrics").returns({error: "Error getting metrics"});
+			const result = index.getCodeqInputsUsingQuery(event, config);
+
+			sinon.assert.calledOnce(getQueryStub);
+			sinon.assert.calledOnce(missingFieldsStub);
+			sinon.assert.calledOnce(fromDateStub);
+			sinon.assert.calledOnce(toDateStub);
+			sinon.assert.calledOnce(fromAfterToStub);
+			sinon.assert.calledOnce(getMetricsStub);
+			
+			expect(result.error).to.eq("Error getting metrics");
+		});
+
+		it('should return data when getCodeq report is called for pathString "codeq"', () => {
+			getMetricsStub = sinon.stub(utils, "getMetrics").returns({metrics:['code-coverage']});
+			const result = index.getCodeqInputsUsingQuery(event, config)
+			
+			sinon.assert.calledOnce(getQueryStub);
+			sinon.assert.calledOnce(missingFieldsStub);
+			sinon.assert.calledOnce(fromDateStub);
+			sinon.assert.calledOnce(toDateStub);
+			sinon.assert.calledOnce(fromAfterToStub);
+			sinon.assert.calledOnce(getMetricsStub);
+			
+			expect(result.metrics[0]).to.eq('code-coverage');
+		});
+	});
+	

```

Comment:
I don't see any unit tests for `getCodeqInputsUsingQuery` specifically for failure scenarios.